src/main/kotlin/platform/mixin/handlers/mixinextras/ModifyExpressionValueHandler.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Minecraft Development for IntelliJ
+ *
+ * https://mcdev.io/
+ *
+ * Copyright (C) 2023 minecraft-dev
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, version 3.0 only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.demonwav.mcdev.platform.mixin.handlers.mixinextras
+
+import com.demonwav.mcdev.platform.mixin.inspection.injector.ParameterGroup
+import com.demonwav.mcdev.util.Parameter
+import com.intellij.psi.PsiAnnotation
+import com.intellij.psi.PsiType
+import org.objectweb.asm.Type
+import org.objectweb.asm.tree.AbstractInsnNode
+import org.objectweb.asm.tree.ClassNode
+import org.objectweb.asm.tree.MethodNode
+
+class ModifyExpressionValueHandler : MixinExtrasInjectorAnnotationHandler() {
+    override val supportedElementTypes = listOf(
+        ElementType.METHOD_CALL, ElementType.FIELD_GET, ElementType.INSTANTIATION, ElementType.CONSTANT
+    )
+
+    override fun extraTargetRestrictions(insn: AbstractInsnNode) =
+        getInsnReturnType(insn)?.equals(Type.VOID_TYPE) == false
```

Comment:
Avoid using `equals` in Kotlin code:
```suggestion
    override fun extraTargetRestrictions(insn: AbstractInsnNode): Boolean {
        val returnType = getInsnReturnType(insn) ?: return false
        return returnType != Type.VOID_TYPE
    }
```
Note that the simpler `getInsnReturnType(insn) != Type.VOID_TYPE` is not equivalent to what you've written, because it returns `true` if `getInsnReturnType(insn)` is `null`.

Reply:
That's why I did it this way in the first place. If your version is deemed to be cleaner then I can change it.